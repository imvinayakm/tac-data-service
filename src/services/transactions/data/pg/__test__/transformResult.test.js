const transformResult = require('../transformResult');

const before = [
  {
    id: 'qwe',
    data_key: 'testint',
    data_type: 'integer',
    data_value_integer: 11,
    data_value_boolean: null,
    data_value_string: null,
    data_value_binary: null,
    position_in_tx: 0,
  },
  {
    id: 'qwe',
    data_key: 'testbool',
    data_type: 'boolean',
    data_value_integer: null,
    data_value_boolean: false,
    data_value_string: null,
    data_value_binary: null,
    position_in_tx: 1,
  },
  {
    id: 'asd',
    data_key: 'testbinary',
    data_type: 'binary',
    data_value_integer: null,
    data_value_boolean: null,
    data_value_string: null,
    data_value_binary: 'base64:qwerqwer',
    position_in_tx: 1,
  },
  {
    id: 'asd',
    data_key: 'teststring',
    data_type: 'string',
    data_value_integer: 11,
    data_value_boolean: null,
    data_value_string: 'some string',
    data_value_binary: null,
    position_in_tx: 0,
  },
];

describe('Data transactions db result transform', () => {
  it('should group raw results by transaction and put nested `data` inside \
    preserving order in position_in_tx field', () => {
    expect(transformResult(before)).toMatchSnapshot();
  });

  it('should handle case when there is no data in tx (all nulls in a row)', () => {
    expect(
      transformResult([
        {
          id: 'asd',
          data_key: null,
          data_type: null,
          data_value_integer: null,
          data_value_boolean: null,
          data_value_string: null,
          data_value_binary: null,
        },
      ])
    ).toMatchSnapshot();
  });

  it('should return empty list for undefined, null or []', () => {
    expect(transformResult()).toEqual([]);
    expect(transformResult(null)).toEqual([]);
    expect(transformResult([])).toEqual([]);
  });
});
