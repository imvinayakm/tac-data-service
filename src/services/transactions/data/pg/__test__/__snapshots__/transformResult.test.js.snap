// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data transactions db result transform should group raw results by transaction and put nested \`data\` inside     preserving order in position_in_tx field 1`] = `
Array [
  Object {
    "data": Array [
      Object {
        "key": "testint",
        "type": "integer",
        "value": 11,
      },
      Object {
        "key": "testbool",
        "type": "boolean",
        "value": false,
      },
    ],
    "id": "qwe",
  },
  Object {
    "data": Array [
      Object {
        "key": "teststring",
        "type": "string",
        "value": "some string",
      },
      Object {
        "key": "testbinary",
        "type": "binary",
        "value": "base64:qwerqwer",
      },
    ],
    "id": "asd",
  },
]
`;

exports[`Data transactions db result transform should handle case when there is no data in tx (all nulls in a row) 1`] = `
Array [
  Object {
    "data": Array [],
    "id": "asd",
  },
]
`;
